src/cets.erl

```diff
@@ -438,7 +438,9 @@ handle_call(get_leader, _From, State = #{leader := Leader}) ->
 handle_call(sync, From, State = #{other_servers := Servers}) ->
     %% Do spawn to avoid any possible deadlocks
     proc_lib:spawn(fun() ->
-        lists:foreach(fun ping/1, Servers),
+        %% If ping crashes, the caller would not receive a reply.
+        %% So, we have to use catch to still able to reply with ok.
+        lists:foreach(fun(Server) -> catch ping(Server) end, Servers),
```

Comment:
So we decide that we still want to return `ok` even though `ping` crashed, which means we might be not fully synced? I want to make sure what the intention is.

Reply:
`sync` is used to ensure that all the servers in the cluster consumed their mail box.
But we only care about messages that were sent before the client code executed `cets:sync/1` (in our case, the "client" code is `cets_join`).

That's actually it, we do not give more guarantees. `ping` here is just a sync gen_server call.
If the connection fails or one of the servers crashes - the code that called sync, could continue execution.
Usually that means that later the local gen_server would receive `DOWN` from the remote gen_server, and could start the cleaning procedure.

Actually, you are right, we could just cancel joining earlier, if any of the servers gets lost. So, we would not need to get the data dump in this case - a bit of optimisation here too.

We could return `{error, {bad_servers, List}}` from sync and of there is an error, `cets_join` would exit earlier (and it would be retried later again).